packages/dds/sequence/src/intervalIndex/overlappingSequenceIntervalsIndex.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	Client,
+	ISegment,
+	ReferenceType,
+	compareReferencePositions,
+} from "@fluidframework/merge-tree";
+import {
+	IIntervalHelpers,
+	IntervalType,
+	SequenceInterval,
+	createPositionReferenceFromSegoff,
+} from "../intervalCollection";
+import { IntervalTree } from "../intervalTree";
+import { SequenceIntervalIndexes } from "./SequenceIntervalIndexes";
+
+class OverlappingSequenceIntervalsIndex implements SequenceIntervalIndexes.Overlapping {
+	private readonly intervalTree = new IntervalTree<SequenceInterval>();
+
+	constructor(
+		private readonly helpers: IIntervalHelpers<SequenceInterval>,
+		private readonly client: Client,
+	) {}
+
+	public remove(interval: SequenceInterval) {
+		this.intervalTree.removeExisting(interval);
+	}
+
+	public add(interval: SequenceInterval) {
+		this.intervalTree.put(interval);
+	}
+
+	public findOverlappingIntervals(
+		startSegoff: { segment: ISegment | undefined; offset: number | undefined },
+		endSegoff: { segment: ISegment | undefined; offset: number | undefined },
+	) {
+		if (this.intervalTree.intervals.isEmpty()) {
+			return [];
+		}
+
+		const startRefPos = createPositionReferenceFromSegoff(
+			this.client,
+			startSegoff,
+			ReferenceType.Transient,
+		);
+
+		const endRefPos = createPositionReferenceFromSegoff(
+			this.client,
+			endSegoff,
+			ReferenceType.Transient,
+		);
+
+		if (compareReferencePositions(startRefPos, endRefPos) > 0) {
+			return [];
+		}
+
+		// initialize a default transient interval
+		const transientInterval = this.helpers.create(
```

Comment:
this defeats the purpose of the index--the idea behind providing this API is that we can implement it without incurring an O(log(# merge tree segments)) cost to locate segment+offset from positions.

Rather than do this, you should create local references directly from the segment+offset, then use those in a SequenceInterval.